transaction/StronglyConnectedComponents.hpp

```diff
@@ -0,0 +1,113 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_TRANSACTION_STRONGLY_CONNECTED_COMPONENTS_HPP_
+#define QUICKSTEP_TRANSACTION_STRONGLY_CONNECTED_COMPONENTS_HPP_
+
+#include <stack>
+#include <vector>
+#include <unordered_map>
+
+#include "transaction/DirectedGraph.hpp"
+#include "utility/Macros.hpp"
+
+namespace quickstep {
+
+namespace transaction {
+
+/** \addtogroup Transaction
+ *  @{
+ */
+
+/**
+ * @brief Class for applying Tarjan's strongly connected components
+ *        algorithm to the 
+ **/
+class StronglyConnectedComponents {
+ public:
+  /**
+   * @brief Constructor for StronglyConnectedComponents.
+   *
+   * @param directed_graph Pointer to the directed graph which the 
+   *        algortihm runs.
+   **/
+  explicit StronglyConnectedComponents(DirectedGraph *directed_graph);
```

Comment:
It is not an owner pointer. I used a pointer to emphasize it. Const reference also makes sense since it never acquires a nullptr. I will change it. Thanks Jignesh.